Overall doc review (#366)
mozilla/firefox-data-docs: src/concepts/analysis_gotchas.md

Most are full timestamps, though their resolution may differ from being per-second to being per-day.

Then there's `profile/creationDate` which is just a number of days since epoch (January 1, 1970).
Like `17177` for the date 2017-01-11.

**Tip:** To convert `profile/creationDate` to a usable date in SQL:
`DATE_ADD('day', profile_created, DATE '1970-01-01')`

In derived datasets ISO dates are sometimes converted to strings in one of
two formats: `%Y-%m-%d` or `%Y%m%d`.

The date formats for different rows in `main_summary` are described on the
[`main_summary` reference page][msref].

Build ids look like dates but aren't.
If you take the first eight characters you can use that as a proxy
for the day the build was released.

`metadata/Date` is an HTTP Date header in a [RFC 7231]-compatible format.

**Tip:** To parse `metadata/Date` to a usable date in SQL:
`DATE_PARSE(SUBSTR(client_submission_date, 1, 25), '%a, %d %b %Y %H:%i:%s')`

[ISO 8601]: https://en.wikipedia.org/wiki/ISO_8601
[msref]: ../datasets/batch_view/main_summary/reference.md#time-formats
[RFC 7231]: http://tools.ietf.org/html/rfc7231#section-7.1.1.1

### Delays

Telemetry data takes a while to get into our hands.

> The largest data mule in Telemetry is the main-ping.

Reviewer: ```suggestion
The bulk of desktop telemetry metrics come through the main-ping, which
```

> It is (pending [Bug 1336360]) sent at the beginning of a client's _next_ Firefox session.

Reviewer: ```suggestion
is (since [Bug 1336360]) sent at the beginning of a client's _next_ Firefox session.
```

If the user shuts down their Firefox for the weekend,
we won't get their Friday data until Monday morning.

A rule of thumb is data from two days ago is usually fairly representative.

If you'd like to read more about this subject and look at pretty graphs,

> there are a series of blog posts [here][delays1], [here][delays2] and [here][delays3].

Reviewer: ```suggestion
there are a series of blog posts ([delays blog post 1][delays1], [delays blog post 2][delays2], [delays blog post 3][delays3]).
```

[Bug 1336360]: https://bugzilla.mozilla.org/show_bug.cgi?id=1336360
[delays1]: https://chuttenblog.wordpress.com/2017/02/09/data-science-is-hard-client-delays-for-crash-pings/
[delays2]: https://chuttenblog.wordpress.com/2017/07/12/latency-improvements-or-yet-another-satisfying-graph/
[delays3]: https://chuttenblog.wordpress.com/2017/09/12/two-days-or-how-long-until-the-data-is-in/

#### Pingsender

Pingsender greatly reduces delay in sending pings to Mozilla,
but only some types of pings are sent by Pingsender.
[Bug 1310703] introduced Pingsender for crash pings and was merged in Firefox 54,
which hit release on June 13, 2017.
[Bug 1336360] moved shutdown pings to Pingsender and was merged in Firefox 55,
which hit release on August 8, 2017.
[Bug 1374270] added sending health pings on shutdown via Pingsender and was
merged in Firefox 56, which hit release on Sept 28, 2017.
Other types of pings are not sent with Pingsender.
This is usually okay because Firefox is expected to continue running long
enough to send those pings.

Mobile clients do not have Pingsender,

> so they suffer delay as given in [this query][delay_q].

Reviewer: ```suggestion
so they suffer delay as given in [this query][delay_q].
See the [Glean overview](glean/glean.md)
for details on ping delay for newer products sending telemetry via Glean.
```
